ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecReducer.java

```diff
@@ -100,20 +101,19 @@ public void configure(JobConf job) {
     isTagged = gWork.getNeedsTagging();
     try {
       keyTableDesc = gWork.getKeyDesc();
-      inputKeyDeserializer = ReflectionUtils.newInstance(keyTableDesc
-          .getDeserializerClass(), null);
-      SerDeUtils.initializeSerDe(inputKeyDeserializer, null, keyTableDesc.getProperties(), null);
+      AbstractSerDe serDe = ReflectionUtils.newInstance(keyTableDesc
+          .getSerDeClass(), null);
+      serDe.initialize(null, keyTableDesc.getProperties(), null);
+      inputKeyDeserializer = serDe;
```

Comment:
Please call this inputKeySerDe, also no need for the interim serde variable.

Reply:
Done.